### PR TITLE
Append playbook to uninstall metrics #2726

### LINKED ISSUE #2726: Suggest: append playbooks or tags to uninstall metrics
https://github.com/openshift/openshift-ansible/blob/master/playbooks/adhoc/metrics_setup/playbooks/install.yml will be deleted, see: #2717

Suggest: append playbooks or tags to uninstall metrics

@vnugent @sdodson

Best regards,
Stéphane

### PR DESCRIPTION
**Pull request by harobed**
Append playbook to uninstall metrics

Fixes #2726

### CODE REVIEW

**Comment by rhcarvalho** [comment]
@harobed thanks for the PR! Could you please update the PR description to include "fixes #\<issue-number\>"?

https://github.com/blog/1506-closing-issues-via-pull-requests

This way, whenever your PR gets merged the issue will be closed automatically.

**Comment by harobed** [comment]
@rhcarvalho Could you please update the PR description to include "fixes #<issue-number>"?

Done :)

**Comment by rhcarvalho** [comment]
The description I see here still says "Append playbook to uninstall metrics #2726", it won't close the issue automatically. Could you please check it again?

File: playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- hosts: oo_first_master
+  tasks:
+  - name: "Remove metrics url from  master config"
+    lineinfile: "state=absent dest=/etc/origin/master/master-config.yaml regexp='^\ \ metricsPublicURL'"
+
+  - name: "Delete metrics objects"
+    command: "{{item}}"
+    with_items:
+      - oc delete all --selector=metrics-infra --namespace openshift-infra --ignore-not-found
+      - oc delete templates --selector=metrics-infra --namespace openshift-infra --ignore-not-found
```
Comment (rhcarvalho) [inline playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml]: tip: you can pass multiple resource types separated by comma:

```
oc delete all,templates --selector=metrics-infra --namespace openshift-infra --ignore-not-found
```

File: playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      - oc delete sa metrics-deployer --namespace openshift-infra --ignore-not-found
+      - oc delete sa hawkular --namespace openshift-infra --ignore-not-found
+      - oc delete sa heapster --namespace openshift-infra --ignore-not-found
+      - oc delete sa cassandra --namespace openshift-infra --ignore-not-found
+      - oc delete secret metrics-deployer --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-cassandra-certificate --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-cassandra-secrets --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-metrics-account --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-metrics-certificate --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-metrics-secrets --namespace openshift-infra --ignore-not-found
```
Comment (rhcarvalho) [inline playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml]: I'm not familiar with these metrics resources, isn't there a label to select them all? If so, I'd suggest a single call to `oc delete`.

```
oc -n openshift-infra delete --ignore-not-found all,templates,sa,secret -l metrics-infra
```

**Comment by rhcarvalho** [comment]
> The description I see here still says "Append playbook to uninstall metrics #2726", it won't close the issue automatically. Could you please check it again?

Aha, I see what happened. The commit message was updated, but not the PR description. I've updated it now.

**Comment by rhcarvalho** [comment]
aos-ci-test

**Comment by harobed** [comment]
> @harobed thanks for the PR! Could you please update the PR description to include "fixes #<issue-number>"?

Fixed

> I have no experience with the metrics setup, but wanted to share some tips wrt oc delete usage.

Fixed

File: playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+- hosts: oo_first_master
+  tasks:
+  - name: "Remove metrics url from  master config"
+    lineinfile: "state=absent dest=/etc/origin/master/master-config.yaml regexp='^\ \ metricsPublicURL'"
```
Comment (rhcarvalho) [inline playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml]: https://github.com/openshift/openshift-ansible/blob/master/docs/best_practices_guide.adoc#Parameters-to-Ansible-modules-SHOULD-use-the-Yaml-dictionary-format-when-3-or-more-parameters-are-being-passed

File: playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+  - name: "Delete metrics objects"
+    command: "{{item}}"
+    with_items:
+      - oc delete all,secrets,templates,sa -l metrics-infra --namespace openshift-infra --ignore-not-found
```
Comment (rhcarvalho) [inline playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml]: With this oneliner I believe we don't need to loop over items?

Comment (rhcarvalho) [inline playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml]: Please replace `oc` with `{{ openshift.common.client_binary }}`

File: playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (rhcarvalho) [inline playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml]: This seems to be missing some things, like removal of role bindings:

- [roles/openshift_metrics/tasks/install.yml#L31-L38](https://github.com/openshift/openshift-ansible/blob/bf0f8266e91b552ffa815d3a7bcb65f105606438/roles/openshift_metrics/tasks/install.yml#L31-L38)
- [roles/openshift_metrics/tasks/install.yml#L49-L56](https://github.com/openshift/openshift-ansible/blob/bf0f8266e91b552ffa815d3a7bcb65f105606438/roles/openshift_metrics/tasks/install.yml#L49-L56)

File: playbooks/byo/openshift-cluster/metrics/uninstall_metrics.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+- hosts: localhost
```
Comment (rhcarvalho) [inline playbooks/byo/openshift-cluster/metrics/uninstall_metrics.yml]: Most yml files in the repo begin with `---`, could you please add that for consistency?

**Comment by rhcarvalho** [comment]
@jcantrill FYI, I think the new metrics playbooks include uninstall?

**Comment by rhcarvalho** [comment]
Here're the new metrics playbooks: https://github.com/openshift/openshift-ansible/pull/2873

@jcantrill, is this PR useful for existing installations?

**Comment by mtnbikenc** [comment]
Closing this PR as it is stale.  Please reopen if you feel this patch is still required.

**Pull request closed without merging**